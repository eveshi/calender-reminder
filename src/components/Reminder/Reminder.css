*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box; }

html {
  font-size: 62.5%; }
  @media only screen and (max-width: 400px) {
    html {
      font-size: 50%; } }

body {
  font-family: 'Roboto', sans-serif;
  background-color: #F5F5F4; }

a svg {
  height: 4rem;
  width: 4rem;
  text-decoration: none;
  fill: #5F513F; }
  a svg:hover {
    background-color: transparent; }

@font-face {
  font-family: 'Roboto', sans-serif;
  src: url("https://fonts.googleapis.com/css?family=Roboto"); }

.Reminder {
  min-height: 32rem;
  border-top: 2px solid #DECDA7;
  display: flex;
  flex-direction: column; }
  .Reminder_freeDay {
    font-size: 5rem;
    text-align: center;
    margin-top: 5rem;
    font-weight: bold;
    color: #BD924B;
    text-shadow: 1px 1px 0.5px #A5A4A2; }
  .Reminder .Button {
    border-bottom: 2px dotted;
    padding-bottom: 0px; }
    .Reminder .Button a {
      display: flex;
      flex-direction: row;
      padding-left: 4rem;
      padding-right: 4rem;
      text-decoration: none;
      color: #060505; }
  .Reminder_ReminderCell_timeString {
    margin-right: 2rem; }
  .Reminder_ReminderCell_reminder {
    width: auto;
    overflow: hidden;
    text-overflow: ellipsis; }
