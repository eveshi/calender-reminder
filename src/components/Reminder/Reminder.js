import React from 'react';
import PropTypes from 'prop-types';
import ReminderCell from '../ReminderCell/ReminderCell';

const Reminder = (props) => {
  const {
    reminders,
  } = props;

  let remindersDisplay = [];

  if (reminders === null) {
    remindersDisplay = [(
      <p key="free">
        You get a free day!
      </p>
    )];
  } else {
    remindersDisplay = reminders.map(reminder => (
      <ReminderCell
        key={reminders.indexOf(reminder)}
        time={reminder.time}
        reminder={reminder.reminder}
      />
    ));
  }

  return (
    <div>
      {remindersDisplay}
    </div>
  );
};

Reminder.propTypes = {
  reminders: PropTypes.arrayOf(PropTypes.object),
};

Reminder.defaultProps = {
  reminders: null,
};

export default Reminder;
