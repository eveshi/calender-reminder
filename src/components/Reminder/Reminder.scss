@import '../../stylesheets/main.scss';

.Reminder {
  min-height: 32rem;
  border-top: 2px solid $color-secondary-middle;
  display: flex;
  flex-direction: column;

  &_freeDay {
    font-size: 5rem;
    text-align: center;
    margin-top: 5rem;
    font-weight: bold;
    color: $color-secondary-dark;
    text-shadow: 1px 1px 0.5px $color-primary-middle;
  }

  .Button {
    border-bottom: 2px dotted;
    padding-bottom: 0px;

    a {
      display: flex;
      flex-direction: row;
      padding-left: 4rem;
      padding-right: 4rem;
      text-decoration: none;
      color: $color-primary-dark;
    }
  }

  &_ReminderCell_timeString {
    margin-right: 2rem;
  }

  &_ReminderCell_reminder {
    width: auto;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}