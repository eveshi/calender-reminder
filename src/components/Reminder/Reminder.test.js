import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Reminder from './Reminder';

configure({ adapter: new Adapter() });

const props = {
  reminders: [],
};

jest.mock('../ReminderCell/ReminderCell.js', () => () => (
  <div id="ReminderCell">
    ReminderCell
  </div>));

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Reminder {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
