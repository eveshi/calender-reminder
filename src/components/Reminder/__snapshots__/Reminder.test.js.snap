// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<div
  className="Reminder"
>
  <p
    className="Reminder_freeDay"
  >
    - FREE DAY -
  </p>
</div>
`;
